File: Skornel02.ETR.Common/Dtos/ProfileDto.cs
using Skornel02.ETR.Common.Dtos;
using Skornel02.ETR.Common.Enums;

namespace Skornel02.ETR.Common.Dtos;

public record ProfileDto
{
    public required string Name { get; set; }
    public required string Username { get; set; }
    public List<DegreeDto> Degrees { get; set; } = [];
    public List<RoleType> Roles { get; set; } = [];
}

File: Skornel02.ETR.Web/src/schemas/ProfileDto.ts
import { z } from 'zod';
import { DegreeDtoSchema } from './DegreeDto';
import { UserType } from '../enums/usertypes';

export const ProfileDtoSchema = z.object({
	name: z.string(),
	username: z.string(),
	degrees: z.array(DegreeDtoSchema),
	roles: z.array(z.nativeEnum(UserType))
});

export type ProfileDto = z.infer<typeof ProfileDtoSchema>;
